### ISSUE TITLE
Elixir maker reports incorrect issues

### ISSUE THREAD

**Issue by drewolson** [neomake/neomake#335]
I tried running the `:Neomake` command on one of my elixir projects that I know to be correct, but neomake is reporting an error on [this line](https://github.com/drewolson/scrivener/blob/01cb9e53ae42a9273b5ec217ef7a87a7d85dc939/lib/scrivener.ex#L47). I've attached a screenshot as well. Perhaps it is having a hard time because `Ecto` is a dependency?

![screen shot 2016-03-05 at 2 33 59 pm](https://cloud.githubusercontent.com/assets/3536/13550277/bb487e10-e2df-11e5-9876-85ba2bd015a4.png)

**Comment by jalcine** [comment]
I'm running into the same problem. I've [created a pull request][1] that _should_ help with this; I just need a bit of help.

[1]: https://github.com/benekastah/neomake/pull/380

**Comment by jalcine** [comment]
@drewolson: Could I lean on some of your experience here with Elixir? I know that `mix` has a compile command we'd want to use here but it seems like the dependencies defined in the `mix.exs` file aren't loaded when [`mix compile` is run][1]. Any thoughts?

[1]: https://github.com/benekastah/neomake/pull/380/files#diff-7694402ada836be51b50f3dfc5209baeR26

**Comment by drewolson** [comment]
@jalcine dependencies are definitely loaded on `mix compile` (I've just confirmed on my project). What behavior are you seeing?

**Comment by jalcine** [comment]
@drewolson: When I use `mix compile.elixir` as the check command, it doesn't remove warnings that are analogous with running `elixir` (like missing dependencies). Example of the code and its dependencies are here: https://gist.github.com/jalcine/b7f85296a0872b81d866353a745b05e1

**Comment by drewolson** [comment]
Try 'mix compile' rather than 'mix compile.elixir'.

**Comment by jalcine** [comment]
@drewolson: Worked like a charm!
@benekastah: I can confirm that the changes in #380 would help resolve this issue. Mind taking a look?

**Comment by dvcrn** [comment]
Giving my +1 here. Out of the box I am facing the same incorrect issues. 

How did you guys get it working?

**Comment by jalcine** [comment]
@dvcrn We got emoji responses for that!

I've personally maintained a local fork that incorporates the changes in #380.

**Comment by dvcrn** [comment]
Are emoji responses / reactions triggering emails? I purposely posted a comment to make it re-appear in the inboxes of people involved and trigger a conversation. The last time I checked #380 (a.k.a. 3 hours ago) it hasn't been updated for a while and had merge conflicts, hence the question if there was maybe a newer solution. 

But I see #380 got updated and merged just now 😶 That was fast! Thanks a lot for the speedy update!
